aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -85,15 +91,55 @@ impl GasAlgebra for StandardGasAlgebra {
         self.balance
     }
 
+    fn check_consistency(&self) -> PartialVMResult<()> {
```

Comment:
@wrwg I wish this is a property I can prove through formal analysis. Do you know if this is possible in Rust?